nilearn/masking.py

```diff
@@ -513,6 +513,137 @@ def compute_multi_background_mask(data_imgs, border_size=2, upper_cutoff=0.85,
     return mask
 
 
+def compute_gray_matter_mask(target_img, threshold=.5,
+                             connected=True, opening=2, memory=None,
+                             verbose=0):
+    """ Compute a mask corresponding to the gray matter part of the brain.
+    The gray matter part is calculated through the resampling of MNI152
+    template Gray Matter mask onto the image
+
+    Parameters
+    ----------
+    target_img: Niimg-like object
+        See http://nilearn.github.io/manipulating_images/input_output.html
+        Images used to compute the mask. 3D and 4D images are accepted.
+        Only the shape and affine of target_img will be used here.
+
+    threshold: float, optional
+        The value under which the MNI template is cut off.
+
+    connected: boolean, optional
+        if connected is True, only the largest connect component is kept.
+
+    opening: bool or int, optional
+        if opening is True, a morphological opening is performed, to keep
+        only large structures.
+        If opening is an integer `n`, it is performed via `n` erosions.
+        After estimation of the largest connected constituent, 2`n` closing
+        operations are performed followed by `n` erosions. This corresponds
+        to 1 opening operation of order `n` followed by a closing operator
+        of order `n`.
+
+    memory: instance of joblib.Memory or string
+        Used to cache the function call.
+
+    verbose: int, optional
+
+    Returns
+    -------
+    mask: nibabel.Nifti1Image
+        The brain mask (3D image)
+    """
+    if verbose > 0:
+        print("Template mask computation")
+
+    target_img = _utils.check_niimg(target_img)
+
+    from .datasets import load_mni152_brain_mask
+    template = load_mni152_brain_mask()
+    dtype = target_img.get_data_dtype()
+    template = new_img_like(template,
+                            template.get_data().astype(dtype))
+
+    from .image.resampling import resample_to_img
+    resampled_template = cache(resample_to_img, memory)(template, target_img)
+
+    mask = resampled_template.get_data() >= threshold
+
+    mask, affine = _post_process_mask(mask, target_img.affine, opening=opening,
+                                      connected=connected,
+                                      warning_msg="Are you sure that input is "
```

Comment:
This warning message needs to be updated according to the target image

Reply:
If I understood correctly, target_img is functional or anatomical ?

Reply:
I need to update this message right, but actually I think that both functional and anatomical images could be used for this resampling